src/main/java/mate/academy/lib/BookDaoImpl.java

```diff
@@ -0,0 +1,112 @@
+                throw new RuntimeException("Expected to insert at leas one ro, but was 0");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(ID_INDEX, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(ID_INDEX, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+                Book book = new Book(title, price);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant create connection to the DB", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM books";
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+        PreparedStatement statement = connection.prepareStatement(sql)) {
+         ResultSet resultSet = statement.executeQuery(sql);
+         while (resultSet.next()) {
+             Long id = resultSet.getObject("id", Long.class);
+             String title = resultSet.getString("title");
+             BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+             Book book = new Book(title, price);
+             books.add(book);
+         }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find books", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setObject(UPDATE_INDEX, book.getId());
+            statement.setString(TITLE_INDEX, book.getTitle());
+            statement.setBigDecimal(PRICE_INDEX, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Expected to insert at leas one row, but was 0");
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book in DB" + book, e);

```

Comment:
don't forget to add space during rows concatenation. 
```suggestion
            throw new DataProcessingException("Can't update book in DB " + book, e);
```